setup.py

```diff
@@ -40,36 +49,135 @@ def find_resources(package_name):
+                package_version,
+            )
+        except ImportError as e:
+            print_error()
+
+    def _build_xacro(self):
+        """ Look for the xacro files and build them in the build folder. """
+        resources_dir = str(
+            Path(__file__).parent.absolute()
+            / "src"
+            / package_name
+            / "resources"
+        )
+        build_folder = str(
+            (
+                Path(self.build_lib) / package_name / "resources" / "urdf"
+            ).absolute()
+        )
+        xacro_files = []
+        for (root, _, files) in walk(str(Path(resources_dir) / "xacro")):
+            for afile in files:
+                if afile.endswith(".urdf.xacro"):
+                    xacro_files.append(str(Path(root) / afile))
+
+        # rebuild all urdfs.
+        rmtree(build_folder, ignore_errors=True)
+        Path(build_folder).mkdir(parents=True, exist_ok=True)
+
+        for xacro_file in xacro_files:
+            for xacro_file in xacro_files:
+                # Generated file name
+                generated_urdf_path = str(
+                    Path(build_folder) / Path(xacro_file).stem
+                )
+                self._build_single_xacro_file(xacro_file, generated_urdf_path)
+
+    def _build_single_xacro_file(self, input_path, output_path):
+        from xacro import process_file, open_output
+        from xacro.xmlutils import xml
+
+        unicode = str
+        encoding = {}
+        print_error(
+            "building xacro file (", input_path, ") into (", output_path, ")"
+        )
+        try:
+            # open and process file
+            doc = process_file(input_path)
+            # open the output file
+            out = open_output(output_path)
+
+        except xml.parsers.expat.ExpatError as e:
+            print_error("XML parsing error: %s" % unicode(e), alt_text=None)
+
+        except Exception as e:
+            msg = unicode(e)
+            if not msg:
+                msg = repr(e)
+            print_error(msg)
+
+        # write output
+        out.write(doc.toprettyxml(indent="  ", **encoding))
+        # only close output file, but not stdout
+        out.close()
+
+    def run(self):
+        """Build the package. """
+        # build documentation.
+        self._build_doc()
+        # build the xacro files into urdf files.
+        self._build_xacro()
+        # distutils uses old-style classes, so no super()
+        build_py.run(self)
+
+
 # Final setup.
 setup(
     name=package_name,
-    version="1.0.0",
+    version=package_version,
     package_dir={package_name: path.join("src", package_name)},
     packages=[package_name],
     package_data={package_name: resources},
+    data_files=data_files_to_install,
     scripts=scripts_list,
     install_requires=[
         "setuptools",
         "xacro",
         "pybullet",
         "importlib_resources",
         "meshcat",
-        "bullet_utils",
```

Comment:
why removing bullet_utils here?

Reply:
because bullet_utils is not on PyPI, including it gives error when installing with pip

Reply:
I disagree, i complains only if bullet_utils is not installed. it can be installed either via colcon(according that your sourced your workspace) or via pip install.